src/tiled/objectsdock.h

```diff
@@ -98,12 +98,18 @@ private slots:
     void onActivated(const QModelIndex &proxyIndex);
     void onSectionResized(int logicalIndex);
     void selectedObjectsChanged();
+    void setColumnVisibility(bool visible);
+
+    void showCustomMenu(const QPoint &point);
 
 private:
+    void updateColumnVisibilityActions();
+    void hideExtraColumns();
     void synchronizeSelectedItems();
 
     MapDocument *mMapDocument;
     QAbstractProxyModel *mProxyModel;
+    QList<QAction*> mActions;
```

Comment:
Please call this `mColumnActions`, just to be a little more clear. However, why keep around the actions at all? You could also just build up the menu in `showCustomMenu` and remove `updateColumnVisibilityActions`.